phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java

```diff
@@ -337,6 +384,22 @@ public boolean nextRaw(List<Cell> result) throws IOException {
                         arrayElementCell = result.get(arrayElementCellPosition);
                     }
                     if (ScanUtil.isLocalIndex(scan) && !ScanUtil.isAnalyzeTable(scan)) {
```

Comment:
We are getting the results after seek only. Will check once again whether we can handle this in local index scanner.